vars/terraform.groovy

```diff
@@ -0,0 +1,177 @@
+/**
+terraform.groovy: Please look at https://github.com/jenkins-infra/shared-tools/tree/main/terraform#jenkins-pipeline for documentation and usage.
+**/
+
+def call(userConfig = [:]) {
+  def defaultConfig = [
+    cronTriggerExpression: '@daily', // Defaults to run once a day (to detect configuration drift)
+    stagingCredentials: [], // No custom secrets for staging by default
+    productionCredentials: [], // No custom secrets for production by default
+    productionBranch: 'main', // Defaults to the default principal branch
+    agentContainerImage: 'jenkinsciinfra/hashicorp-tools:0.3.5', // version is managed by updatecli
+    runTests: false, // execute the tests provided by the "calling" project, that should provide a tests/Makefile
```

Comment:
```suggestion
    runTests: false, // Executes the tests provided by the "calling" project, which should provide a tests/Makefile
```

Reply:
Does it need to be explicited? Can't this pipeline looks for the presence of test file in the repo instead?

Reply:
maybe it could. But I prefer to make it customizable by the end user: too much "magic" is a pain for me. If you feel it is important, please feel free to PR :)